lib/req_ch.ex

```
@@ -128,13 +160,21 @@ defmodule ReqCH do
   end
 
   defp add_format(%Req.Request{} = request) do
-    format = request |> Req.Request.get_option(:format, :tsv) |> normalise_format!()
+    format_option = Req.Request.get_option(request, :format, :tsv)
+    format = normalise_format(format_option)
 
-    format_header = with :explorer <- format, do: "Parquet"
+    if format do
+      format_header = with :explorer <- format, do: "Parquet"
 
-    request
-    |> Req.Request.put_private(:clickhouse_format, format)
-    |> Req.Request.put_header("x-clickhouse-format", format_header)
+      request
+      |> Req.Request.put_private(:clickhouse_format, format)
+      |> Req.Request.put_header("x-clickhouse-format", format_header)
+    else
+      Req.Request.halt(
+        request,
+        format_error(format_option)
+      )
```

I think this is probably fine but honestly I'd just raise if format is invalid. The way I try to think about it is if user supplies wrong options we want to raise, if the server responds with garbage (e.g. invalid JSON) we gracefully handle that with returning exception struct. (Not that the user can do much with it _anyway_ but people wanted that so they got it.)

---

Yeah, I was unsure about this, so I changed to raise instead. Thanks!